bindinfo/handle.go

```diff
@@ -569,7 +569,7 @@ func (h *BindHandle) CaptureBaselines() {
 func getHintsForSQL(sctx sessionctx.Context, sql string) (string, error) {
 	oriVals := sctx.GetSessionVars().UsePlanBaselines
 	sctx.GetSessionVars().UsePlanBaselines = false
-	recordSets, err := sctx.(sqlexec.SQLExecutor).ExecuteInternal(context.TODO(), fmt.Sprintf("explain format='hint' %s", sql))
+	recordSets, err := sctx.(sqlexec.SQLExecutor).Execute(context.TODO(), fmt.Sprintf("explain format='hint' %s", sql))
```

Comment:
Why?

Reply:
`ExecuteInternal` will allow SQL hint see the invisible index. But here we do not want SQL hint see the invisible index. For example,
When enable `@@TiDBCapturePlanCache` and execute `admin capture bindings`, TiDB will execute an internal SQL `explain format='hint' <origin sql>` in here. It will see all invisible index and may bind SQL with SQL hint includes invisible index. It's a wrong behavior, The easiest fix way is let it become a non-internal execute.

Reply:
This change may impact the logging / metrics of these internal queries, it may lead to confusing outcomes. You'd better either to add another flag variable, or to allow baseline capture to see those invisible indexes, because even if the bind contains invisible index, it would not take effect when using the bind. @zz-jason any thoughts on the behavior here?

Reply:
@eurekaka Comment addressed, introduce a new session variable `OptimizerUseInvisibleIndexes`, PTAL